includes/bp-messages/classes/class-bp-rest-sitewide-notices-endpoint.php

```diff
@@ -0,0 +1,1000 @@
+<?php
+/**
+ * BP REST: BP_REST_Sitewide_Notices_Endpoint class
+ *
+ * @package BuddyPress
+ * @since 9.0.0
+ */
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Sitewide Notices endpoints.
+ *
+ * @since 9.0.0
+ */
+class BP_REST_Sitewide_Notices_Endpoint extends WP_REST_Controller {
+
+	/**
+	 * Constructor.
+	 *
+	 * @since 9.0.0
+	 */
+	public function __construct() {
+		$this->namespace = bp_rest_namespace() . '/' . bp_rest_version();
+		$this->rest_base = 'sitewide-notices';
+	}
+
+	/**
+	 * Register the component routes.
+	 *
+	 * @since 9.0.0
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base,
+			array(
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_items' ),
+					'permission_callback' => array( $this, 'get_items_permissions_check' ),
+					'args'                => $this->get_collection_params(),
+				),
+				array(
+					'methods'             => WP_REST_Server::CREATABLE,
+					'callback'            => array( $this, 'create_item' ),
+					'permission_callback' => array( $this, 'create_item_permissions_check' ),
+					'args'                => $this->get_endpoint_args_for_item_schema( WP_REST_Server::CREATABLE ),
+				),
+				'schema' => array( $this, 'get_item_schema' ),
+			)
+		);
+
+		// (?P<id>[\d]+) is the placeholder for the notice ID.
+		$notice_endpoint = '/' . $this->rest_base . '/(?P<id>[\d]+)';
+
+		register_rest_route(
+			$this->namespace,
+			$notice_endpoint,
+			array(
+				'args' => array(
+					'id' => array(
+						'description' => __( 'A unique numeric ID for the Sitewide notice.', 'buddypress' ),
+						'type'        => 'integer',
+					),
+				),
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_item' ),
+					'permission_callback' => array( $this, 'get_item_permissions_check' ),
+					'args'                => $this->get_endpoint_args_for_item_schema( WP_REST_Server::READABLE ),
+				),
+				array(
+					'methods'             => WP_REST_Server::EDITABLE,
+					'callback'            => array( $this, 'update_item' ),
+					'permission_callback' => array( $this, 'update_item_permissions_check' ),
+					'args'                => $this->get_endpoint_args_for_item_schema( WP_REST_Server::EDITABLE ),
+				),
+				array(
+					'methods'             => WP_REST_Server::DELETABLE,
+					'callback'            => array( $this, 'delete_item' ),
+					'permission_callback' => array( $this, 'delete_item_permissions_check' ),
+					'args'                => $this->get_endpoint_args_for_item_schema( WP_REST_Server::DELETABLE ),
+				),
+				'schema' => array( $this, 'get_item_schema' ),
+			)
+		);
+
+		$dismiss_endpoint = '/' . $this->rest_base . '/dismiss';
+
+		register_rest_route(
+			$this->namespace,
+			$dismiss_endpoint,
```

Comment:
Any reason to not do this?
```suggestion
		register_rest_route(
			$this->namespace,
			'/' . $this->rest_base . '/dismiss',
```

Reply:
I don't think so, I agree & I've applied your suggestion 👍